client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -21,32 +22,167 @@ const EditProfilePage = () => {
     }
   }, [profile]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
     const { name, value } = e.target;
     setFormData((prevFormData) => ({
       ...prevFormData,
       [name]: value,
     }));
   };
+  const handleCompanyChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      careerInformation: {
+        ...prevData.careerInformation,
+        currentPosition: {
+          ...prevData.careerInformation.currentPosition,
+          company: value,
+        },
+      },
+    }));
+  };
+  
+ const [skillInput, setSkillInput] = useState("");
+
+  const handleSkillChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSkillInput(e.target.value);
+  };
+
+  const handleSpecialization = (skill: string) => {
+    setFormData((prevData) => {
+      const isAlreadySpecialized = prevData.specializations.includes(skill);
+      return isAlreadySpecialized
+        ? prevData
+        : {
+            ...prevData,
+            specializations: [...prevData.specializations, skill],
+          };
+    });
+  };
+
+  const handleSkillKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && skillInput.trim() !== "") {
+      e.preventDefault();
+      if (!formData.skills.includes(skillInput.trim())) {
+        setFormData((prevData) => ({
+          ...prevData,
+          skills: [...prevData.skills, skillInput.trim()],
+        }));
+      }
+      setSkillInput("");
+    }
+  };
+
+  const handleSkillRemove = (skill: string, specialty: boolean) => {
+    if (specialty) {
+      setFormData((prevData) => ({
+        ...prevData,
+        specializations: prevData.specializations.filter((s) => s !== skill),
+      }));
+    } else {
+      setFormData((prevData) => ({
+        ...prevData,
+        skills: prevData.skills.filter((s) => s !== skill),
+      }));
+    }
+  };
+
+  // Handling the change for title
+  const handleTitleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      careerInformation: {
+        ...prevData.careerInformation,
+        currentPosition: {
+          ...prevData.careerInformation.currentPosition,
+          title: value,
+        },
+      },
+    }));
+  };
 
+  const handleTwitterChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      socialMediaLinks: {
+        ...prevData.socialMediaLinks,
+        twitter: value,
+      },
+    }));
+  };
+
+  const handleBlogChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      socialMediaLinks: {
+        ...prevData.socialMediaLinks,
+        blog: value,
+      },
+    }));
+  };
```

Comment:
i think there's a way to consolidate `handleTwitterChange` and `handleBlogChange` since they're both changing state on the same nested property